hexapod_dart/src/hexapod_dart_simu.cpp

```diff
@@ -334,3 +291,86 @@ void HexapodDARTSimu::_add_floor()
 
     _world->addSkeleton(floor);
 }
+
+double HexapodDARTSimu::_min_dist_angle(double a1, double a2)
```

Comment:
There's a check afterwards if the result is exceeding 180 (or below -180).

So, for your example: -190+360 = 170.